lib/tasks/e2e-test.js

```diff
@@ -85,32 +83,29 @@ const configure = function configure (gulp, opts, env) {
       ]);
     }
 
-    // go through the steps to run the test
-    return startupSeq.reduce((promise, step) => promise.then(step), B.resolve())
-      .then(function () {
-        return globby(e2eTestFiles);
-      })
-      .then(function (files) {
-        // gulp-mocha has an issue where, if there are no files passed from gulp.src,
-        // it will just run everything it finds
-        if (!files.length) {
-          return;
-        }
-        return mochaCmd();
-      })
-      .then(function () {
-        return cleanupSeq.reduce((promise, step) => promise.then(step), B.resolve())
-          .finally(function () {
-            if (opts.e2eTest.forceExit) {
-              process.exit(0);
-            }
-          });
-      }).catch(function (err) {
-        return cleanupSeq.reduce((promise, step) => promise.then(step), B.resolve())
-          .finally(function () {
-            env.spawnWatcher.handleError(err);
-          });
-      });
+    try {
+      // go through the steps to run the test
+      await startupSeq.reduce((promise, step) => promise.then(step), B.resolve());
+      const files = await globby(e2eTestFiles);
+      // gulp-mocha has an issue where, if there are no files passed from gulp.src,
+      // it will just run everything it finds
+      if (!files.length) {
+        log(`No e2e test files found using '${e2eTestFiles}'`);
+        return;
+      }
+      await mochaCmd();
+      await cleanupSeq.reduce((promise, step) => promise.then(step), B.resolve());
+    } catch (err) {
+      try {
+        await cleanupSeq.reduce((promise, step) => promise.then(step), B.resolve());
```

Comment:
shouldn't this be executed in finally block?

Reply:
It cannot. The error handler exits the process, so the overall finally block is never run.